polyglot/piranha/src/models/rule.rs

```diff
@@ -242,29 +255,29 @@ impl Rule {
     );
 
     // Return the first match that satisfies constraint of the rule
-    for (range, tag_substitutions) in all_query_matches {
+    for (replacement_range, matches) in all_query_matches {
       let matched_node = get_node_for_range(
         source_code_unit.root_node(),
-        range.start_byte,
-        range.end_byte,
+        replacement_range.start_byte,
+        replacement_range.end_byte,
       );
       if matched_node.satisfies_constraint(
         source_code_unit.clone(),
         self,
-        &tag_substitutions
+        &matches
           .clone()
           .into_iter()
           .chain(rule_store.input_substitutions())
           .collect(),
         rule_store,
       ) {
-        let replacement = substitute_tags(self.replace(), &tag_substitutions).replace("\\n", "\n");
-        return Some(Edit::new(
-          range,
-          replacement,
-          self.clone(),
-          tag_substitutions,
-        ));
+        if applied_edit_start.is_none()
```

Comment:
Ensure that the next reported match occurs before (line-column number wise) the previously reported match.